go/pkg/rexec/rexec.go

```diff
@@ -196,9 +196,16 @@ func (ec *Context) GetCachedResult() {
 		ec.resPb = resPb
 	}
 	if ec.resPb != nil {
+		ec.Result = command.NewResultFromExitCode((int)(ec.resPb.ExitCode))
+		ec.setOutputMetadata()
 		cmdID, executionID := ec.cmd.Identifiers.ExecutionID, ec.cmd.Identifiers.CommandID
 		log.V(1).Infof("%s %s> Found cached result, downloading outputs...", cmdID, executionID)
-		ec.Result = ec.downloadResults(ec.cmd.ExecRoot, ec.opt.DownloadOutputs)
+		if ec.opt.DownloadOutErr {
+			ec.Result = ec.downloadOutErr()
```

Comment:
We need to save the failure result if we fail to download anything, including outerr. I suggest maybe a different order (output files first, because they are more important?) -- and if they failed, don't attempt the other download. Or maybe save the error anyway even if the other download succeeds.
Or maybe use your DownloadOutputs/Stderr functions to preserve the status.

Reply:
Good point. I went with cancelling the second download, and kept OutErr first so its consistent with the current behavior.